plugins/providers/shield/resource.go

```diff
@@ -0,0 +1,157 @@
+package shield
+
+import (
+	"fmt"
+
+	"github.com/odpf/guardian/domain"
+)
+
+const (
+	ResourceTypeTeam         = "team"
+	ResourceTypeProject      = "project"
+	ResourceTypeOrganization = "organization"
+)
+
+type Metadata struct {
+	Email   string `json:"email" mapstructure:"email"`
+	Privacy string `json:"privacy" mapstructure:"privacy"`
+	Slack   string `json:"slack" mapstructure:"slack"`
+}
+
+type User struct {
+	ID    string `json:"id" mapstructure:"id"`
+	Name  string `json:"name" mapstructure:"name"`
+	Email string `json:"email" mapstructure:"email"`
+}
+
+type Team struct {
+	ID       string   `json:"id" mapstructure:"id"`
+	Name     string   `json:"name" mapstructure:"name"`
+	Slug     string   `json:"slug" mapstructure:"slug"`
+	OrgId    string   `json:"orgId" mapstructure:"orgId"`
+	Metadata Metadata `json:"metadata" mapstructure:"metadata"`
+	Admins   []string `json:"admin" mapstructure:"admin"`
+}
+
+type Project struct {
+	ID     string   `json:"id" mapstructure:"id"`
+	Name   string   `json:"name" mapstructure:"name"`
+	Slug   string   `json:"slug" mapstructure:"slug"`
+	OrgId  string   `json:"orgId" mapstructure:"orgId"`
+	Admins []string `json:"admin" mapstructure:"admin"`
+}
+
+type Organization struct {
+	ID     string   `json:"id" mapstructure:"id"`
+	Name   string   `json:"name" mapstructure:"name"`
+	Slug   string   `json:"slug" mapstructure:"slug"`
+	Admins []string `json:"admin" mapstructure:"admin"`
+}
+
+func (t *Team) FromDomain(r *domain.Resource) error {
+	if r.Type != ResourceTypeTeam {
+		return ErrInvalidResourceType
+	}
+
+	resourceDetails := r.Details
+	t.ID = resourceDetails["id"].(string)
+	t.OrgId = resourceDetails["orgId"].(string)
```

Comment:
would these details be always populated? if not, it will panic.

Reply:
Id can't be empty and orgId can be empty as it is extra info.

Reply:
Are we trying to cast interface to string? This could throw panic if the interface is not parsable to string. Can you make it more defensive by adding?
```go
t.ID, ok := resourceDetails["id"].(string)
...
t.OrgId, ok  := resourceDetails["orgId"].(string)
...
```
[ref](https://go.dev/ref/spec#Type_assertions)